src/js/plugin/sdk/pipeline/PipelineModel.js

```diff
@@ -0,0 +1,48 @@
+
+class PipelineModel {
+
+  constructor() {
+    this.pipelines = new Map();
+  };
+
+  clearPipeline() {
+    this.pipelines.clear();
+  };
+
+  applyFilter(operationName, filterParameters) {
+    const pipeline = this.getPipeline(operationName);
+
+    pipeline.forEach((filter) => {
```

Comment:
Let's rewrite this method to properly pass the input and output parameter so that `applyPipeline(FOO, "bar") === c(b(a("bar"))))` given we've registered the functions `a`, `b`, `c` as operators for "FOO".

```js
function a(input){
 return input.toUpperCase();
}

function b(input){
 return input.split("").reverse().join("");
}

function c(input){
 return input.replace(/[AEUIO]/g,"");
}

pipelines =  new PipelineModel();
pipelines.registerOperator("FOO", a); 
pipelines.registerOperator("FOO", b); 
pipelines.registerOperator("FOO", c);

pipelines.applyPipeline("FOO", "bar") === c(b(a("bar")))
```

https://github.com/mesosphere/marathon-ui/pull/819#issuecomment-352718900 illustrates how to implement such a method.

Reply:
:+1: